src/main/java/com/marianbastiurea/lifeofbees/Queen.java

```diff
@@ -30,46 +40,61 @@ public int getAgeOfQueen() {
     public void setAgeOfQueen(int ageOfQueen) {
         this.ageOfQueen = ageOfQueen;
     }
-    //    public void makeEggs(Hive hive, Environment environment) {
-//        int numberOfEggs = (int) (2000 */* honeyIndex * */environment.getWeather().getTemperature() * ageOfQueenIndex(ageOfQueen));
-//
-//        hive.addEggs(numberOfEggs);
-//    }
 
-    public double ageOfQueenIndex() {
-        Queen queen = new Queen();
-
-        /* a queen lives 3-5 years. We will consider 1 years old for our queen. At age 5,
-    beekeper will have to replace this queen with new one.Depending on age of queen will choose an fertility index
-     between 0 and 1. When index is 0, queen is too old to lay eggs and she have to be replaced. Whenn index is 1,
+    public double ageOfQueenIndex(Hive hive) {
+       // Hive hive = new Hive();
+        /* a queen lives 3-5 years. When will build first 10 hives in apiary will use random to generate ageOfQueen
+         between 1  and 5 years old for our queen. At age 5, will have to replace this queen with new one.
+         Depending on age of queen will choose an fertility index between 0 and 1. When index is 0, queen is too old
+         to lay eggs and she have to be replaced. Whenn index is 1,
      fertility of queen is at maximum and she can lay upon 2000 eggs daily
          */
 
-        int ageOfQueen = queen.getAgeOfQueen();
+        int ageOfQueen =hive.getAgeOfQueen();
         double numberRandom = Math.random();
         switch (ageOfQueen) {
             case 0, 1, 2, 3:
                 return 1;
             case 4:
                 if (numberRandom < 0.5) {
-                    queen.setAgeOfQueen(0);
+                    hive.setAgeOfQueen(0);
                     return 1;
                 } else
                     return 0.75;
             case 5:
-                queen.setAgeOfQueen(0);
+                hive.setAgeOfQueen(0);
                 return 0.25;
             default:
                 break;
         }
         return 0;
     }
 
-    public static int makeEggs() {
+    public int makeEggs(Hive hive) {
         Queen queen = new Queen();
-        int numberOfEggs = (int) (2000 * queen.ageOfQueenIndex());
-        // have to add another index, a  honeyIndex which will depend on quantity of honey made it
+        int numberOfEggs = (int) (2000 * queen.ageOfQueenIndex(hive));//*Whether.geWhetherIndex());
```

Comment:
Probably you want to use `this` instead of instantiating a new queen.

Reply:
I want to use queen of each hive. in fact I need ageOfQueenIndex for each ageOfQueen from each hive.

Reply:
Because `ageOfQueenIndex` method only uses a hive object, the `ageOfQueenIndex` method should be part of `Hive` class. `ageOfQueen` method an property should be property of `Queen` class not of `Hive` class. If you need the age of a queen of a Hive you should access it through the Queen example  `hive.getQueen().getAgeOfQueen()`

Reply:
done